gobblin-service/src/main/java/org/apache/gobblin/service/modules/dataset/ExternalDatasetDescriptor.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.service.modules.dataset;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigValueFactory;
+import java.io.IOException;
+import java.util.ArrayList;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import org.apache.gobblin.service.modules.flowgraph.DatasetDescriptorConfigKeys;
+import org.apache.gobblin.service.modules.flowgraph.DatasetDescriptorErrorUtils;
+import org.apache.gobblin.util.ConfigUtils;
+
+
+/**
+ * Describes a external dataset not on HDFS
+ * e.g, https://some-api:443/user/123/names where the path is the full URI
```

Comment:
the concept of 'external' is vague (and only defined by negation--"not on HDFS").  is it about being remote?  is it about HTTP?  if so, must it be static (e.g. REST) or do we also cover hosted services (API invocations)?

...or is this more just a generic container meant to be open-ended and flexible?

Reply:
It's meant to be a generic container that's open ended

Reply:
I stil find this description unclear, given we already have `SqlDatasetDescriptor`s and `IcebergDatasetDescriptor`s, which likewise are "not on HDFS".

part of my confusion would be resolved by renaming and part by more clearly documenting what one if these *is* (rather than what it is not).

e.g. does `GenericWebDatasetDescriptor` (or `GenericHttpDatasetDescriptor`) match your intent?

Reply:
No because this can also apply to SFTP and SQL flows running on https://github.com/linkedin/data-integration-library/blob/master/docs/parameters/summary.md